libexec/bats-core/bats

```diff
@@ -93,6 +93,7 @@ expand_path() {
 BATS_LIBEXEC="$(dirname "$(expand_link "${BASH_SOURCE[0]}")")"
 export BATS_CWD="$PWD"
 export BATS_TEST_PATTERN="^[[:blank:]]*@test[[:blank:]]+(.*[^[:blank:]])[[:blank:]]+\{(.*)\$"
+export BATS_TEST_PATTERN_COMMENT="[[:blank:]]*([^[:blank:]()]+)[[:blank:]]*\(?\)?[[:blank:]]*\{[[:blank:]]*#[[:blank:]]*@test[[:blank:]]*\$"
```

Comment:
Just a little nitpick, following invalid patterns are possible with this pattern:

- `should_be_found) {# @test`
- `should_be_found( {# @test`

Reply:
Additionally, could users expect to be able to write `function function_name() { #test`?

Reply:
> following invalid patterns are possible with this pattern

I agree they would be allowed with this pattern, but the point of this pattern is that _valid bash works_, and neither of the patterns listed are valid bash.  We could make the regex require `[:blank:]` before and after the `{` which would cause these 2 examples to not match, which in turn would cause an error when the test file was run:

```text
 ✗ test comment style
   (in test file test/bats.bats, line 619)
     `[ $status -eq 0 ]' failed
   [1] <<<1..6 /tmp/bats-run-9663/bats.9682.src: line 25: syntax error near unexpected token `)' /mnt/c/Users/ltheisen/git/pastdev-bats-core/lib/bats-core/tracing.bash: line 61: BATS_STACK_TRACE: bad array subscript not ok 1 setup_file failed # bats warning: Executed 1 instead of expected 6 tests>>>
```

But I don't think that's necessary.  I don't know of a way to prevent `(` or `)` without a non-capturing group because `BASH_REMATCH[2]` is used to match the body so if i try `(\(\))?` to enforce `()` or nothing, then it is used as the body of the function and fails.  Thus, it would be a much more extensive modification.  In my opinion, we should probably leave the regex as is and leave it up to the users to ensure their code is valid bash.

> Additionally, could users expect to be able to write function function_name() { #@test? It might be very confusing if these are not translated into tests.

I just [added some tests](https://github.com/bats-core/bats-core/pull/191/commits/ddd169d296fc0efa080968de9565f3ad6c225fab) that demonstrates that `function name()` style functions will indeed be identified as tests.  While not explicitly supported (according to bash `help function`) it does seem to be allowed bash syntax, so seems like they should be found as tests.

Reply:
> the point of this pattern is that valid bash works

You're right, I was scanning the regex for completeness and made a list that contained more items but after realising that the others were wrong I posted just those two, hence the "nitpick" part.

> I just added some tests that demonstrates that function name() style functions will indeed be identified as tests.

Oh, I just noticed you already had tests for `function name {` which I actually was getting at, it was not so much about the parens but them working too won't hurt.